### PR TITLE
PMM-3981 Fix reset offset on slow log rotation.

### PR DESCRIPTION
**Pull request by askomorokhov**
https://jira.percona.com/browse/PMM-3981

Steps to reproduce:

0) setup pmm-server, mysql with enabled slow log and sys bench 
1) add mysql with slowlog 
```bash
pmm-agent setup --config-file=/usr/local/percona/pmm-agent.yaml --server-address=pmm-server:443 --server-insecure-tls
pmm-agent --config-file=/usr/local/percona/pmm-agent.yaml &
pmm-admin add mysql  --username=root --password=secret --use-slowlog ps:3306 MySQLSlowLog
```
3) wait few minutes to get some buckets from slowlog.
4) check how many buckets in ClickHouse 
```bash
docker exec pmm-server clickhouse client -d pmm -q "select count(*) from metrics;"
```
5) rotate slowlog
```
 mysql  -e "show variables like '%slow_query%';"
 mysql  -e "set global slow_query_log=off;"
 mysql  -e "flush slow logs;"
mv /var/log/mysql/mysql-slow.log /var/log/mysql/mysql-slow-$(date +%Y-%m-%dT%H:%M:%S).log
 mysql  -e "set global slow_query_log=on;"
```
6) 3) wait few minutes to get some buckets from new slowlog.
7) check if added new buckets in ClickHouse 
```bash
docker exec pmm-server clickhouse client -d pmm -q "select count(*) from metrics;"
```

### CODE REVIEW

File: agents/builtin/mysql/slowlog/slowlog.go [hunk 1]
```diff
@@ -148,7 +149,7 @@ func (m *SlowLog) Run(ctx context.Context) {
 				// FIXME https://jira.percona.com/browse/PMM-3898
 				m.l.Infof("Events channel is closed.")
 				running = false
-				m.changes <- Change{Status: inventorypb.AgentStatus_WAITING}
+				m.changes <- Change{Status: inventorypb.AgentStatus_RUNNING}
```
Comment (AlekSi) [inline agents/builtin/mysql/slowlog/slowlog.go]: My understand is that we are waiting for a restart there. Why RUNNING? Also, that does not make sense to do that right after `running = false` assignment.

File: agents/builtin/mysql/slowlog/slowlog.go [hunk 2]
```diff
@@ -100,12 +101,12 @@ func (m *SlowLog) Run(ctx context.Context) {
 		return
 	}
 
-	stat, err := os.Stat(slowLogFilePath)
+	operatingSlowLogFileStat, err := os.Stat(slowLogFilePath)
```
Comment (AlekSi) [inline agents/builtin/mysql/slowlog/slowlog.go]: `stat`

File: agents/builtin/mysql/slowlog/slowlog.go [hunk 3]
```diff
@@ -175,17 +176,20 @@ func (m *SlowLog) Run(ctx context.Context) {
 			res := aggregator.Finalize()
 
 			// Check if MySQL SlowLog config is changed and slowlog rotated.
-			curStat, err := os.Stat(slowLogFilePath)
+			newSlowLogFileStat, err := os.Stat(slowLogFilePath)
```
Comment (AlekSi) [inline agents/builtin/mysql/slowlog/slowlog.go]: `curStat` or `newStat`. https://github.com/golang/go/wiki/CodeReviewComments#variable-names `newSlowLogFileStat` is a super long name for a local variable in Go.

File: agents/builtin/mysql/slowlog/slowlog.go [hunk 4]
```diff
@@ -175,17 +176,20 @@ func (m *SlowLog) Run(ctx context.Context) {
...
+			// In case of rotatation a slowlog file, set a new offset to parse the new slow log file.
+			if !os.SameFile(operatingSlowLogFileStat, newSlowLogFileStat) {
+				opts.StartOffset = uint64(newSlowLogFileStat.Size())
+				// use new slowlog stat as current.
+				operatingSlowLogFileStat = newSlowLogFileStat
 			}
 
 			if !running {
-				m.changes <- Change{Status: inventorypb.AgentStatus_STARTING}
+				m.changes <- Change{Status: inventorypb.AgentStatus_RUNNING}
```
Comment (AlekSi) [inline agents/builtin/mysql/slowlog/slowlog.go]: `STARTING` is for when we started doing something, but not received any data yet. Looks like STARTING is a correct value there.

File: agents/builtin/mysql/slowlog/slowlog_test.go [hunk 5]
```diff
@@ -17,9 +17,55 @@
 package slowlog
 
 import (
+	"encoding/json"
+	"io/ioutil"
 	"testing"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
```
Comment (AlekSi) [inline agents/builtin/mysql/slowlog/slowlog_test.go]:
```suggestion
	"github.com/golang/protobuf/proto"
```

Comment (askomorokhov) [reply agents/builtin/mysql/slowlog/slowlog_test.go]: Fixed.

File: agents/builtin/mysql/slowlog/slowlog.go (hunk 4 above)
Comment (askomorokhov) [reply agents/builtin/mysql/slowlog/slowlog.go]: Yeah, agree.

File: agents/builtin/mysql/slowlog/slowlog.go (hunk 3 above)
Comment (askomorokhov) [reply agents/builtin/mysql/slowlog/slowlog.go]: Renamed.

File: agents/builtin/mysql/slowlog/slowlog.go (hunk 2 above)
Comment (askomorokhov) [reply agents/builtin/mysql/slowlog/slowlog.go]: renamed.

**Pull request merged**